src/ucp/api/ucp.h

```diff
@@ -732,7 +764,25 @@ typedef enum {
      * data by calling @ref ucp_am_recv_data_nbx routine. This flag is mutually
      * exclusive with @a UCP_AM_RECV_ATTR_FLAG_DATA.
      */
-    UCP_AM_RECV_ATTR_FLAG_RNDV         = UCS_BIT(17)
+    UCP_AM_RECV_ATTR_FLAG_RNDV          = UCS_BIT(17),
+
+    /**
+     * Indicates that the incoming data is the first fragment of the
+     * multi-fragment eager message. This flag can only be passed to data
+     * handlers registered without @a UCP_AM_FLAG_WHOLE_MSG flag. This flag is
+     * mutually exclusive with @a UCP_AM_RECV_ATTR_FLAG_RNDV and
+     * @a UCP_AM_RECV_ATTR_FLAG_ONLY flags.
+     */
+    UCP_AM_RECV_ATTR_FLAG_FIRST         = UCS_BIT(18),
+
+    /**
+     * Indicates that the incoming data carries the whole message. This flag is
+     * mutually exclusive with @a UCP_AM_RECV_ATTR_FLAG_RNDV and
+     * @a UCP_AM_RECV_ATTR_FLAG_FIRST flags. Also this flags is always passed to
+     * the data handlers, which are registered with @a UCP_AM_FLAG_WHOLE_MSG
+     * flag.
+     */
+    UCP_AM_RECV_ATTR_FLAG_ONLY          = UCS_BIT(19)
```

Comment:
maybe `FLAG_FRAG_SINGLE`?

Reply:
is it really needed, can just compare length < total_length?

Reply:
in this case we will always have to set total_length (even for tiny messages).
This flag is proposed for convenient (one branch) check for multi-fragment vs sinlgle fragment message